service/share/full_availability.go

```diff
@@ -0,0 +1,31 @@
+package share
+
+import (
+	"context"
+
+	format "github.com/ipfs/go-ipld-format"
+
+	"github.com/celestiaorg/celestia-node/ipld"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+// fullAvailability implements Availability using the full data square
+// recovery technique. It is considered "full" because it is required
+// to download enough shares to fully reconstruct the data square.
+type fullAvailability struct {
+	getter format.NodeGetter
+}
+
+// NewFullAvailability creates a new full Availability.
+func NewFullAvailability(get format.NodeGetter) Availability {
+	return &fullAvailability{
+		getter: get,
+	}
+}
+
+// SharesAvailable reconstructs the data committed to the given Root by requesting
+// enough Shares from the network.
+func (fa *fullAvailability) SharesAvailable(ctx context.Context, root *Root) error {
```

Comment:
nit: Is `SharesAvailable` a good name for a method that reconstructs the block? Is this because both "availability" implementations share the same interface or sth?

Reply:
Yes but we're gonna think about the `Availability` interface a bit and how it interacts w/ ShareService soon. It's for another PR tho.

Reply:
cc @Wondertan

Reply:
`SharesAvailable` checks if shares are available. 'FullAvailability' implies full availability, which involves reconstruction.